ApiBundle/Controller/NodeController.php

```diff
@@ -66,11 +68,18 @@ public function showOrCreateAction(Request $request, $nodeId, $errorNode)
         $language = $request->get('language', $currentSiteDefaultLanguage);
         $siteId = $currentSiteManager->getCurrentSiteId();
         $node = $this->findOneNode($nodeId, $language, $siteId, $request->get('version'));
-
+        if (!$errorNode && $node) {
```

Comment:
I think the ``showOrCreate`` method is growing dramatically which is not good for readability. Can use some protected functions call to make it cleraer ? Maybe one for the  ``if (!$errorNode && $node)`` and one for if (!$node)